mrmustard/training/parametrized.py

```diff
@@ -54,9 +59,34 @@ def __init__(self, **kwargs):  # NOTE: only kwargs so that we can use the arg na
             bounds = kwargs.get(f"{name}_bounds", None)
             param = create_parameter(value, name, is_trainable, bounds, owner)
 
-            # dynamically assign variable as attribute of the class
+            # dynamically assign parameter as attribute of the class
             self.__dict__[name] = param
-            self.param_names[k] = name
+            self.param_names.append(name)
```

Comment:
If using the keys of the `kwargs` dict this line can be removed
```suggestion
```

Reply:
see above